thread.c

```diff
@@ -629,6 +631,17 @@ static void thread_libevent_process(evutil_socket_t fd, short which, void *arg)
     }
 }
 
+// Interface is slightly different on various platforms.
+// On linux, at least, the len limit is 16 bytes.
+#define THR_NAME_MAXLEN 16
+void thread_setname(pthread_t thread, const char *name) {
+assert(strlen(name) < THR_NAME_MAXLEN);
```

Comment:
Might want to copy the assert to the implementation in extstore.c as well?

Reply:
Thanks. I'll re-copy-paste it when I decide if I'm going to move the function or not